src/metabase/activity_feed/models/recent_views.clj

```diff
@@ -572,7 +563,7 @@
      :dashboard  (m/index-by :id (dashboard-recents dashboard-ids))
      :collection (m/index-by :id (collection-recents collection-ids))
      :table      (m/index-by :id (table-recents table-ids))
-     :document   (m/index-by :id (select-documents-for-recents document-ids))}))
+     :document   (m/index-by :id ((requiring-resolve 'metabase.documents.recent-views/select-documents-for-recents) document-ids))}))
```

Comment:
if this is beacuse of cyclic deps I'd move the implementation of `select-documents-for-recents` to here instead.